src/main/java/com/minecolonies/entity/ai/citizen/guard/EntityAIRangeGuard.java

```diff
@@ -0,0 +1,300 @@
+
+        //Lower the variable higher the chance that the arrows hits the target.
+        double chance = HIT_CHANCE_DIVIDER / (worker.getExperienceLevel() + 1);
+
+        arrowEntity.setThrowableHeading(xVector, yVector + distance * AIM_SLIGHTLY_HIGHER_MULTIPLIER, zVector, (float) ARROW_SPEED, (float) chance);
+
+        addEffectsToArrow(arrowEntity, baseDamage);
+
+        worker.addExperience(XP_EACH_ARROW);
+        worker.faceEntity(entityToAttack, (float) TURN_AROUND, (float) TURN_AROUND);
+        worker.getLookHelper().setLookPositionWithEntity(entityToAttack, (float) TURN_AROUND, (float) TURN_AROUND);
+
+        double xDiff = targetEntity.posX - worker.posX;
+        double zDiff = targetEntity.posZ - worker.posZ;
+
+        double goToX = xDiff > 0 ? MOVE_MINIMAL : -MOVE_MINIMAL;
+        double goToZ = zDiff > 0 ? MOVE_MINIMAL : -MOVE_MINIMAL;
+
+        worker.moveEntity(goToX, 0, goToZ);
+
+        worker.swingArm(EnumHand.MAIN_HAND);
+        worker.playSound(SoundEvents.ENTITY_SKELETON_SHOOT, (float) BASIC_VOLUME, (float) getRandomPitch());
+        worker.worldObj.spawnEntityInWorld(arrowEntity);
+
+        worker.damageItemInHand(1);
+    }
+
+    private int getReloadTime()
+    {
+        return BASE_RELOAD_TIME / (worker.getExperienceLevel() + 1);
+    }
+
+    /**
+     * Method used to add potion/enchantment effects to the bow depending on his enchantments etc.
+     *
+     * @param arrowEntity the arrow to add these effects to.
+     * @param baseDamage  the arrow base damage.
+     */
+    private void addEffectsToArrow(EntityTippedArrow arrowEntity, double baseDamage)
+    {
+        int powerEntchantment = EnchantmentHelper.getMaxEnchantmentLevel(Enchantments.POWER, worker);
+        int punchEntchantment = EnchantmentHelper.getMaxEnchantmentLevel(Enchantments.PUNCH, worker);
+
+        DifficultyInstance difficulty = this.worker.worldObj.getDifficultyForLocation(new BlockPos(worker));
+        arrowEntity.setDamage((baseDamage * BASE_DAMAGE_MULTIPLIER)
+                                + worker.getRandom().nextGaussian() * RANDOM_DAMAGE_MULTPLIER
+                                + this.worker.worldObj.getDifficulty().getDifficultyId() * DIFFICULTY_DAMAGE_INCREASE);
+
+        if (powerEntchantment > 0)
+        {
+            arrowEntity.setDamage(arrowEntity.getDamage() + (double) powerEntchantment * BASE_POWER_ENCHANTMENT_DAMAGE + POWER_ENCHANTMENT_DAMAGE_MULTIPLIER);
+        }
+
+        if (punchEntchantment > 0)
+        {
+            arrowEntity.setKnockbackStrength(punchEntchantment);
+        }
+
+        boolean onFire = worker.isBurning() && difficulty.func_190083_c() && worker.getRandom().nextBoolean();
+        onFire = onFire || EnchantmentHelper.getMaxEnchantmentLevel(Enchantments.FLAME, worker) > 0;
+
+        if (onFire)
+        {
+            arrowEntity.setFire(FIRE_EFFECT_CHANCE);
+        }
+
+        ItemStack holdItem = worker.getHeldItem(EnumHand.OFF_HAND);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'holdItem' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)